pandas/core/accessors.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+
+An example/recipe for creating a custom accessor.
```

Comment:
I'm +1 on the general idea of exposing a way to register custom accessors, though the api in your example seems a little clunky compared to what xarray does?  That said, may be better to punt that piece to a follow-up PR.